crates/aptos-profiler/src/memory_profiler.rs

```diff
@@ -2,29 +2,92 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use anyhow::Result;
-use crate::MemProfilerConfig;
-use crate::Profiler;
+use crate::{MemProfilerConfig, Profiler, utils::convert_svg_to_string};
+use std::{thread, fs, path::{Path, PathBuf}, time::Duration, process::Command};
 
 pub struct MemProfiler {
+    duration: u64,
+    memory_profiling_data_file: PathBuf,
 }
 
 impl MemProfiler {
     pub(crate) fn new(config: &MemProfilerConfig) -> Self {
         Self {
+            duration: config.duration,
+            memory_profiling_data_file: config.mem_profiling_data_files_dir.clone(),
+        
         }
     }
 }
 
 impl Profiler for MemProfiler {
     fn start_profiling(&self) -> Result<()> {
-        unimplemented!()
+        unsafe {
+            let mut prof_active: bool = true;
+    
+            let result = jemalloc_sys::mallctl(
```

Comment:
does it work if only change the environment variable to start profiling? or we have to use unsafe here?